addons/stock/models/stock_move_line.py

```diff
@@ -92,6 +107,14 @@ def _compute_picking_type_id(self):
             if line.picking_id:
                 line.picking_type_id = line.picking_id.picking_type_id
 
+    @api.depends('move_id', 'move_id.location_id', 'move_id.location_dest_id')
+    def _compute_location_id(self):
+        for line in self:
+            if not line.location_id:
+                line.location_id = line.move_id.location_id or line.picking_id.location_id
+            if not line.location_dest_id:
+                line.location_dest_id = line.move_id.location_dest_id or line.picking_id.location_id
```

Comment:
@beledouxdenis I have a question: why do you check like that? When I change the target position on stock picking, it will not change the stock move line, leading to the stock quants being in the wrong position, making it difficult for users to understand. Can you explain to me why? Thank you